lib/resty/redis.lua

```diff
@@ -63,7 +64,13 @@ function _M.new(self)
     if not sock then
         return nil, err
     end
-    return setmetatable({ _sock = sock, _subscribed = false }, mt)
+    return setmetatable({ _sock = sock,
+                          _subscribed = false,
+                          _n_channel = {
```

Comment:
I personally prefer adding two keys instead of a table.

Reply:
Use table make code like `if self._n_channel[cmd] == 0 then` more clear and safer to me, since we limit the scope can be changed.